Twitter.agent.lib.nut

```diff
@@ -120,25 +148,41 @@ class Twitter {
                 // Expected status code
                 // Note '23' accompanies an over-large anomalous data block from Twitter
                 // Try again immediatly:
-                imp.wakeup(0, function() { stream(searchTerms, onTweet, onError); }.bindenv(this));
+                _reconnectTimer = imp.wakeup(0,
+                    function() {
+                        stream(searchTerms, onTweet, onError);
+                    }.bindenv(this)
+                );
             } else if (resp.statuscode == 420 || resp.statuscode == 429) {
                 // Too many requests
                 // Try again with the _reconnectTimeout
-                imp.wakeup(_reconnectTimeout, function() { stream(searchTerms, onTweet, onError); }.bindenv(this));
-                _reconnectTimeout *= 2;
+                _reconnectTimer = imp.wakeup(_reconnectTimeout,
+                    function() {
+                        stream(searchTerms, onTweet, onError);
+                    }.bindenv(this)
+                );
+                _reconnectTimeout += TWITTER_DEFAULT_RECONNECT_TIMEOUT_SEC;
```

Comment:
Right, and I believe it should be linear rather then exponential. But I may be wrong.